### PR TITLE
rename DoubleSum -> Sum

### PR DESCRIPTION
**Pull request by codeboten**
**Description:**
Continuing the work started in updating OTLP to v0.8.0, `DoubleSum` is now `Sum`

**Link to tracking Issue:** Part of #3534

### CODE REVIEW

**Review by bogdandrutu — COMMENTED** [review]
I think we should split this:
1. Changes in cmd/mdatagen
2. Changes in pdata (and their usages)
3. Changes in internal/goldendata

If possible :)

**Comment by bogdandrutu** [comment]
After looking at the PR only changes in the  internal/goldendata about the pict files worries me a bit, if these can be separated and discuss more we can move forward with what we have here

**Comment by codeboten** [comment]
> After looking at the PR only changes in the internal/goldendata about the pict files worries me a bit, if these can be separated and discuss more we can move forward with what we have here

Happy to remove the changes to pict files, are there steps to regenerate them automatically documented anywhere?

**Comment by bogdandrutu** [comment]
@codeboten I am useless here, see https://github.com/open-telemetry/opentelemetry-collector/issues/3378 where I asked the same :))

**Comment by codeboten** [comment]
fwiw, I ran

```
pict ./testdata/pict_input_metrics.txt > testdata/generated_pict_pairs_metrics.txt 
```

With the data in this PR and the output matched this PR. Note that there is an existing discrepancy in the data fixed in #3587

File: internal/goldendataset/pict_metrics_gen.go
Comment (bogdandrutu) [inline internal/goldendataset/pict_metrics_gen.go:77]: Ok my point was that I don't think we should do this change since, the Sum will have 2 possible values (Int or Double). So for this is good to keep Double in the name to signal that Sum will have Double values in the NumberDataPoint.

Comment (codeboten) [reply internal/goldendataset/pict_metrics_gen.go:77]: Ah I see. Thanks for clarifying!

Comment (codeboten) [reply internal/goldendataset/pict_metrics_gen.go]: pict changes reverted.

**Comment by bogdandrutu** [comment]
Unfortunately I cannot merge this because the upgrade of the contrib for otel-trace changes is blocked. To not waste time let's start address other issues in the meantime.

One PR that we can do is to have the conversion of IntHistogram to Histogram. We need to call this code every time when we unmarshal metrics so all the components will be able to ignore IntHistograms and we can delete them.

I am proposing this PR until we unblock the contrib upgrade. What do you think?

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
LGTM but blocking because of contrib not being able to be upgraded because of otel-trace issues.

**Comment by codeboten** [comment]
> One PR that we can do is to have the conversion of IntHistogram to Histogram. We need to call this code every time when we unmarshal metrics so all the components will be able to ignore IntHistograms and we can delete them.

Sounds like a good next step @bogdandrutu, I've also got the PRs to rename `DoubleGauge` ready to go once the `DoubleSum` PRs are merged.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
